README.md

```diff
@@ -32,22 +33,231 @@ scheduled_task { 'Run Notepad':
+
+###### win32_taskscheduler
+
+This provider manages scheduled tasks on Windows using the legacy API.
+
+###### taskscheduler_api2
+
+This provider manages scheduled tasks on Windows using the v2 api.
+
+* Default for `operatingsystem` == `windows`.
+
+##### `trigger`
+
+One or more triggers defining when the task should run.
+A single trigger is represented as a hash, and multiple triggers can be specified with an array of hashes.
+
+A trigger can contain the following keys:
+
+For all triggers:
+
+* `schedule` (Required) — What kind of trigger this is.
+  Valid values are `daily`, `weekly`, `monthly`, or `once`.
+  Each kind of trigger is configured with a different set of keys; see the sections below (once triggers only need a start time/date.)
+* `start_time` (Required) — The time of day when the trigger should first become active.
+  Several time formats will work, but we suggest 24-hour time formatted as HH:MM.
+* `start_date` — The date when the trigger should first become active.
+  Defaults to the current date.
+  You should format dates as YYYY-MM-DD, although other date formats may work (under the hood, this uses Date.parse).
+* `minutes_interval` — The repeat interval in minutes.
+* `minutes_duration` — The duration in minutes, needs to be greater than the minutes_interval.
+* For daily triggers:
+  * `every` — How often the task should run, as a number of days.
+    Defaults to 1.
+    “2” means every other day, “3” means every three days, etc.
+* For weekly triggers:
+  * `every` — How often the task should run, as a number of weeks.
+    Defaults to 1.
+    “2” means every other week, “3” means every three weeks, etc.
+  * `day_of_week` — Which days of the week the task should run, as an array.
+    Defaults to all days.
+    Each day must be one of `mon`, `tues`, `wed`, `thurs`, `fri`, `sat`, `sun`, or `all`.
+* For monthly (by date) triggers:
+  * `months` — Which months the task should run, as an array.
+    Defaults to all months.
+    Each month must be an integer between 1 and 12.
+  * `on` (Required) — Which days of the month the task should run, as an array.
+    Each day must be an integer between 1 and 31.
+* For monthly (by weekday) triggers:
+  * `months` — Which months the task should run, as an array. Defaults to all months. Each month must be an integer between 1 and 12.
+  * `day_of_week` (Required) — Which day of the week the task should run, as an array with only one element.
+    Each day must be one of `mon`, `tues`, `wed`, `thurs`, `fri`, `sat`, `sun`, or `all`.
+  * `which_occurrence` (Required) — The occurrence of the chosen weekday when the task should run. Must be one of first, second, third, fourth, or fifth.
+
+##### `user`
+
+The user to run the scheduled task as.
+Please note that not all security configurations will allow running a scheduled task as 'SYSTEM', and saving the scheduled task under these conditions will fail with a reported error of 'The operation completed successfully'.
```

Comment:
Is that right? That doesn't sound much like an error message?

Reply:
That's pulled straight from existing docs. 😢

Reply:
There are some weird behaviors in the v1 API... that hopefully are not the same in the v2 API.

Here is the original comment from @ferventcoder that led to the above - https://github.com/puppetlabs/puppet/pull/2848#issuecomment-49052377
It would make sense to try out his same manifest to see if the behavior is the same.

@glennsarti left a similar comment at https://github.com/puppetlabs/puppet/pull/5042/files#diff-414807a601087eb6bf7e40a0aa442471R229

https://github.com/puppetlabs/puppet/pull/5035/commits also has some details around how `save` worked in v1.